inference-engine/src/gna_plugin/frontend/scale_factor_calc.hpp

```diff
@@ -1180,6 +1181,36 @@ template<>
 class ScaleFactorPerLayer<InferenceEngine::ConvolutionLayer*> : public ScaleFactorPerLayer<InferenceEngine::WeightableLayer*> {
 };
 
+template<>
+class ScaleFactorPerLayer<InferenceEngine::GemmLayer*> {
+public:
+    bool operator() (InferenceEngine::GemmLayer* gemmLayer, int weightsSize, int inputSize, ScaleFactorUpdateResult &result, const bool fakeQuantize) {
+        if ( !gemmLayer ) {
+            THROW_GNA_EXCEPTION << "Incorrect Gemm Layer pointer \n";
+        }
+        auto in0 = InferenceEngine::CNNNetPrevLayer(gemmLayer, 0);
+        auto in1 = InferenceEngine::CNNNetPrevLayer(gemmLayer, 1);
+
+        auto quantData = InferenceEngine::getInjectedData<QuantizedLayerParams>(*gemmLayer);
+        auto quantParams1 = InferenceEngine::getInjectedData<QuantizedLayerParams>(in1);
+        auto quantParams0 = InferenceEngine::getInjectedData<QuantizedLayerParams>(in0);
+        quantData->_src_quant.SetScale(quantParams0->_dst_quant.GetScale());
+        quantData->_weights_quant.SetScale(quantParams1->_dst_quant.GetScale());
+        if (quantData->_src_quant.IsStatsSet()) {
+            auto getScale = [&quantParams0](size_t i) {
+                return (quantParams0->_dst_quant.GetLevels() - 1) /
+                       (quantParams0->_dst_quant.GetMaxValues(false)[i] -
+                        quantParams0->_dst_quant.GetMinValues(false)[i]);
+            };
+            float min_channel_scale = getScale(0);
+            quantParams0->_dst_quant.SetScale(min_channel_scale);
+            quantData->_src_quant.SetScale(min_channel_scale);
+        }
+        quantData->_dst_quant.SetScale(
```

Comment:
should we also check Gemm destination quant statistics and reduce weights scale factor in the case if int overflow is possible? (as it's done for weightable layers)